src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -3500,8 +3519,9 @@ opendir(const char *path)
 	struct ht_fd      *fd_ht_obj   = NULL;
 
 	if (next_opendir == NULL) {
-		next_opendir = dlsym(RTLD_NEXT, "opendir");
-		D_ASSERT(next_opendir != NULL);
+		query_real_dlsym();
+		next_opendir = real_dlsym(RTLD_NEXT, "opendir");
+		assert(next_opendir != NULL);
```

Comment:
why not D_ASSERT?

Reply:
opendir() could be called before the constructor function is executed, so the logging utility may be not initialized yet.
This PR was developed to work around the bugs inside dlio_benchmark's dependent libraries.
It looks like the issue is resolved in the latest dlio_benchmark. I will wait for Mohamad's confirmation after he is back. We may do not need this patch anymore.
Thank you very much!